### PR TITLE
Making client-side negotiation more pluggable.

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

**Comment by nmittler** [comment]
@ejona86 PTAL

File: netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java [hunk 1]
```diff
@@ -415,7 +445,7 @@ private static boolean installJettyTlsProtocolSelection(final SSLEngine engine,
         final Method removeMethod = negoClass.getMethod("remove", SSLEngine.class);
         putMethod.invoke(null, engine, Proxy.newProxyInstance(
             null,
-            new Class[] {server ? serverProviderClass : clientProviderClass},
+            new Class[]{server ? serverProviderClass : clientProviderClass},
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java]: It is supposed to have the space between ] and {.

File: netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java [hunk 2]
```diff
@@ -97,51 +93,27 @@
   private boolean terminated;
 
   NettyClientTransport(SocketAddress address, Class<? extends Channel> channelType,
-      NegotiationType negotiationType, EventLoopGroup group, SslContext sslContext,
-      int connectionWindowSize, int streamWindowSize) {
-    Preconditions.checkNotNull(negotiationType, "negotiationType");
+                       EventLoopGroup group, ProtocolNegotiator negotiator,
+                       int connectionWindowSize, int streamWindowSize) {
+    Preconditions.checkNotNull(negotiator, "negotiator");
+    Preconditions.checkNotNull(negotiator, "negotiator");
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java]: This second check seems a bit too thorough :-)

**Comment by ejona86** [comment]
@nmittler LGTM. Two small comments.

File: netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java (hunk 2 above)
Comment (nmittler) [inline netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java]: LMAO ... fine!!!! ;)

File: netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java (hunk 1 above)
Comment (nmittler) [inline netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java]: Strange that checkstyle didn't find this ... done.

**Comment by nmittler** [comment]
cherry-picked as 65769c2

**Pull request closed without merging**